vars/piperPipelineStageInit.groovy

```diff
@@ -125,7 +123,7 @@ void call(Map parameters = [:]) {
 
         if (config.legacyConfigSettings) {
             Map legacyConfigSettings = readYaml(text: libraryResource(config.legacyConfigSettings))
```

Comment:
You could also move the readYaml call into the step

Reply:
This has implications for unit tests, not sure if it is worth it